### DISCUSSION TITLE
Cannot train coqui tts glow model for English

### DISCUSSION THREAD

**Discussion by Saad20x** [coqui-ai/TTS#2636] [General Q&A]
Hello,

I wanted to ask if anyone could help me out, I am following the notebook of the coqui tts training model and I follow each and every but the output voice is just static noise..

I thought that it was the dataset issue, but I used the ljspeech direct download, extract and use method.

I didn't change a single cell,

Also can anyone tell me the format of ljspeech csv is 

I tried

Audio1      transcription
Audio1|    transcription
Audio1|transcription
Audio1|transcription|transcription

Apologies for inconvenience I'm a nervous beginner.

Kind Regards

**Comment by p0p4k** [comment]
Hello, you need to give us more information to work with if you want help. What is the error?

**Reply by Saad20x** [reply]
Hi,

So the error is that, the code runs till end, but in the end there isn't any speech in output, its just some static noise....

i followed this tutorial:

https://github.com/coqui-ai/TTS/blob/dev/notebooks/Tutorial_2_train_your_first_TTS_model.ipynb

**Reply by p0p4k** [reply]
If you are training a new model from scratch, increase the number of epochs and train it. If you are training a LJspeech model, just download the pretrained one and do inference on it.

**Reply by Saad20x** [reply]
**I'm training a new model from scratch for english dataset,:
the dataset is ljspeech:
13100 audio wavs.** 

**this is the model config:**
from TTS.tts.configs.glow_tts_config import GlowTTSConfig
config = GlowTTSConfig(
    batch_size=32,
    eval_batch_size=16,
    num_loader_workers=4,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=100,
    text_cleaner="phoneme_cleaners",
    use_phonemes=True,
    phoneme_language="en-us",
    phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    print_step=25,
    print_eval=False,
    mixed_precision=True,
    output_path=output_path,
    datasets=[dataset_config],
    save_step=1000,
)



**This is the audio config file:**

from TTS.utils.audio import AudioProcessor
ap = AudioProcessor.init_from_config(config)

**Reply by p0p4k** [reply]
epochs=100, <-- increase to 500. (try to visualize on tensorboard and see the training status with some generated samples as well)
Also, would suggest you to train using a py file instead of notebook, in future. Right now for testing it's ok.

**Reply by Saad20x** [reply]
@p0p4k Thank you very much for the answer, it was helpful as the speech was detected after 50 epochs,

I actually have the model saved as 
best_model012.pth
config.json

I want to start training again on this model and train on further epochs on the model that was saved, How do I exactly do that, How do I load the model?

import os

# BaseDatasetConfig: defines name, formatter and path of the dataset.
from TTS.tts.configs.shared_configs import BaseDatasetConfig

output_path = "tts_train_dir"
if not os.path.exists(output_path):
    os.makedirs(output_path)
    

!wget -O $output_path/LJSpeech-1.1.tar.bz2 https://data.keithito.com/data/speech/LJSpeech-1.1.tar.bz2 
!tar -xf $output_path/LJSpeech-1.1.tar.bz2 -C $output_path

dataset_config = BaseDatasetConfig(
    formatter="ljspeech", meta_file_train="metadata.csv", path=os.path.join(output_path, "LJSpeech-1.1/")
)


from TTS.tts.configs.glow_tts_config import GlowTTSConfig
config = GlowTTSConfig(
    batch_size=32,
    eval_batch_size=16,
    num_loader_workers=4,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=100,
    text_cleaner="phoneme_cleaners",
    use_phonemes=True,
    phoneme_language="en-us",
    phoneme_cache_path=os.path.join(output_path, "phoneme_cache"),
    print_step=25,
    print_eval=False,
    mixed_precision=True,
    output_path=output_path,
    datasets=[dataset_config],
    save_step=1000,
)



from TTS.utils.audio import AudioProcessor
ap = AudioProcessor.init_from_config(config)



from TTS.tts.utils.text.tokenizer import TTSTokenizer
tokenizer, config = TTSTokenizer.init_from_config(config)


from TTS.tts.datasets import load_tts_samples
train_samples, eval_samples = load_tts_samples(
    dataset_config,
    eval_split=True,
    eval_split_max_size=config.eval_split_max_size,
    eval_split_size=config.eval_split_size,
)


from TTS.tts.models.glow_tts import GlowTTS
model = GlowTTS(config, ap, tokenizer, speaker_manager=None)


from trainer import Trainer, TrainerArgs
trainer = Trainer(
    TrainerArgs(), config, output_path, model=model, train_samples=train_samples, eval_samples=eval_samples
)



trainer.fit()



model_path = '/content/drive/MyDrive/tts_train_dir/LJSpeech-1.1/best_model_20300.pth'
config_path = '/content/drive/MyDrive/tts_train_dir/LJSpeech-1.1/config.json'


How do I use the model and config file from paths and utilize it in the above code? as there are many errors being thrown up,

**Reply by p0p4k** [reply]
Read the documentation and use search bar in GitHub issued and discussions. You will find lot of repetitive threads in there. If things are harder, post again. You can join a very active discord group as well. Don't be nervous, we are here to help. Good luck!

**Reply by Saad20x** [reply]
@p0p4k Thank you for the answer it was really helpful, one thing i wanted to ask is that after 250 epochs the voice output sounds like a computer, will 500 epochs make any difference or should 1000 epochs be used?

**Reply by p0p4k** [reply]
Check your loss graph on tensorboard and see if the model has converged or yet to converge and then make a decision.
